config_builder.go

```diff
@@ -785,28 +785,60 @@ func (d *RPCSignerWalletImpl) BuildChainControl(
 	partialChainControl *chainreg.PartialChainControl,
 	walletConfig *btcwallet.Config) (*chainreg.ChainControl, func(), error) {
 
+	// Keeps track of both the remote signer and the chain control clean up
+	// functions.
+	var (
+		cleanUpTasks []func()
+		cleanUp      = func() {
+			for _, fn := range cleanUpTasks {
+				if fn == nil {
```

Comment:
When would a cleanup function be nil? Shouldn't we be able to expect them always to be non-nil, the way we add them below?

Reply:
Yeah that's correct, currently they'll never be nil given the current code. This was a copy of the pattern used in the `BuildWalletConfig` function, and I guess it could be useful if we'd ever add a remote signer implementation that doesn't have any cleanup func.

But I'll remove the `nil` check if you think that's just adding extra complexity :)!